pkg/status/binding.go

```diff
@@ -42,6 +49,82 @@ func (c *Controller) syncBinding(ctx context.Context, key string) error {
 		c.workqueue.AddAfter(combinedStatusRef(combinedStatus.ObjectName.AsNamespacedName().String()), queueingDelay)
 	}
 
+	bp, err := c.bindingPolicyLister.Get(key)
+	if err != nil {
+		return fmt.Errorf("failed to get BindingPolicy %s from cache: %w", key, err)
+	}
+	missingStatusCollectors, err := c.identifyMissingStatusCollectors(bp)
```

Comment:
This is still working too hard. It is not necessary to lookup the StatusCollector objects. Each `combinedStatusResolution` already knows which StatusCollectors it is missing; it is those for which `combinedStatusResolution.StatusCollectorNameToData[collectorName]` is `nil`.

Reply:
Done.